docs/concepts/controllers.md

```diff
@@ -0,0 +1,111 @@
+The typical package structure for the controllers that are part of etcd-druid is shown with the *custodian controller*:
+
+``` bash
+controllers/custodian
+├── config.go
+├── reconciler.go
+└── register.go
+```
+
+- `config.go`: contains all the logic for the configuration of the controller, including feature gate activations, CLI flag parsing and validations.
+- `register.go`: contains the logic for registering the controller with the etcd-druid controller manager.
+- `reconciler.go`: contains the controller reconciliation logic.
+
+Each controller package also contains auxiliary files which are relevant to that specific controller.
+
+## Controller Manager
+
+A *manager* is first created for all controllers that are a part of etcd-druid.
+The *controller manager* is responsible for all the controllers that are associated with CRDs.
+Once the manager is `Start()`-ed, all the controllers that are *registered* with it are started.  
+
+Each controller is built using a controller builder, configured with details such as the type of object being reconciled, owned objects whose owner object is reconciled, event filters (predicates), etc. `Predicates` are filters which allow controllers to filter which type of events the controller should respond to and which ones to ignore.
+
+The logic relevant to the controller manager like the creation of the controller manager and registering each of the controllers with the manager, is contained in [`controllers/manager.go`](/controllers/manager.go).
+
+## Etcd Controller
+
+The *etcd controller* is responsible for the reconciliation of the `Etcd` resource.
+It handles the provisioning and management of the etcd cluster. Different components that are required for the functioning of the cluster like `Leases`, `ConfigMap`s, and the `Statefulset` for the etcd cluster are all deployed and  managed by the *etcd controller*.
+
+While building the controller, an event filter is set such that the behaviour of the controller depends on the `gardener.cloud/operation: reconcile` *annotation*. This is controlled by the `--ignore-operation-annotation` CLI flag, which, if set to `false`, tells the controller to perform reconciliation only when this annotation is present. If the flag is set to `true`, the controller will trigger reconciliation anytime the `Etcd` spec, and thus `generation`, changes.  
+
+The reason this filter is present is because any disruption in the `Etcd` resource due to reconciliation (due to changes in the `Etcd` spec, for example) while workloads are being run would be disastrous.
+Hence, any user who wishes to avoid such disruptions, can choose to set the `--ignore-operation-annotation` CLI flag to `false`. An example of this is Gardener's [gardenlet](https://github.com/gardener/gardener/blob/master/docs/concepts/gardenlet.md), which reconciles of the `Etcd` resource only during a shoot cluster's [*maintenance window*](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_maintenance.md).
```

Comment:
```suggestion
Hence, any user who wishes to avoid such disruptions, can choose to set the `--ignore-operation-annotation` CLI flag to `false`. An example of this is Gardener's [gardenlet](https://github.com/gardener/gardener/blob/master/docs/concepts/gardenlet.md), which reconciles the `Etcd` resource only during a shoot cluster's [*maintenance window*](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_maintenance.md).
```

Reply:
Thanks for the correction!

Reply:
Addressed in [a210894](https://github.com/gardener/etcd-druid/pull/722/commits/a2108945df415036f1622186141bddf4c74288b1)